internal/handshake/crypto_setup_server.go

```diff
@@ -23,6 +23,8 @@ type KeyExchangeFunction func() crypto.KeyExchange
 
 // The CryptoSetupServer handles all things crypto for the Session
 type cryptoSetupServer struct {
+	mutex sync.RWMutex
```

Comment:
This mutex is never used. Should it be, if users can now make calls into the crypto setup?

Reply:
I just moved that line up. The mutex is used a bunch of times, for example for opening and sealing packets.
We need to lock it for `ConnectionState` calls, in all crypto setups. I'll update the PR.

Reply:
Sorry, didn't catch that. LGTM.

Reply:
Done.